hail/src/main/scala/is/hail/expr/ir/lowering/LowerTableIR.scala

```diff
@@ -54,6 +54,46 @@ object TableStage {
   def wrapInBindings(body: IR, letBindings: IndexedSeq[(String, IR)]): IR = letBindings.foldRight[IR](body) {
     case ((name, value), body) => Let(name, value, body)
   }
+
+  def concatenate(ctx: ExecuteContext, children: IndexedSeq[TableStage]): TableStage = {
+    val keyType = children.head.kType
+    assert(keyType.size == 0)
```

Comment:
Sure. Currently that is already checked in the TableUnion constructor, but doesn't hurt to check again here, in case this ever gets called from elsewhere.